src/main/java/com/vention/stock_market_share/controller/StockController.java

```diff
@@ -17,15 +17,16 @@ public StockController(AlphaVintageService alphaVintageService) {
         this.alphaVintageService = alphaVintageService;
     }
 
-    @GetMapping("/searchStockBySymbol")
-    public ResponseEntity<AlphaVintageResponse> searchStockBySymbol(@RequestParam String symbol, @RequestParam String apiKey) {
+    @GetMapping("/{symbol}")
+    public ResponseEntity<AlphaVintageResponse> searchStockBySymbol(@PathVariable String symbol, @Value("${alphavantage.api.key}") String apiKey) {
```

Comment:
We don't need to carry @Value("${alphavantage.api.key}") through the entire application. Apply it in AlphaVintageClient